### PR TITLE
test integration: force kill etcd if it hangs

### PR DESCRIPTION
**Pull request by aojea**
/kind failing-test
```release-note
NONE
```

avoid waiting forever if etcd hangs
ref: https://github.com/kubernetes/kubernetes/issues/107857

### CODE REVIEW

File: test/integration/framework/etcd.go [hunk 1]
```diff
@@ -129,8 +129,17 @@ func RunCustomEtcd(dataDir string, customFlags []string) (url string, stopFn fun
...
-		klog.Infof("etcd exit status: %v", err)
+		errCh := make(chan error, 1)
+		go func() {
+			errCh <- cmd.Wait()
+		}()
+		select {
+		case err := <-errCh:
+			klog.Infof("etcd exit status: %v", err)
+		case <-time.After(60 * time.Second):
+			err := cmd.Process.Kill()
```
Comment (aojea) [inline test/integration/framework/etcd.go:140]: hmm the context kills the process too , but at least this wait we are not blocked by wait

File: test/integration/framework/etcd.go [hunk 2]
```diff
@@ -129,8 +129,17 @@ func RunCustomEtcd(dataDir string, customFlags []string) (url string, stopFn fun
 	cmd.Stderr = os.Stderr
 	stop := func() {
 		cancel()
-		err := cmd.Wait()
-		klog.Infof("etcd exit status: %v", err)
+		errCh := make(chan error, 1)
```
Comment (liggitt) [inline test/integration/framework/etcd.go:132]: Is this where we're hanging?

Comment (liggitt) [reply test/integration/framework/etcd.go:132]: This is the stop function called when a package is done running tests, right? I thought we were hanging in the middle of particular test functions

Comment (aojea) [reply test/integration/framework/etcd.go:132]: hmm, so it is possible that we never get here and we get stuck in one test?

Comment (aojea) [reply test/integration/framework/etcd.go:132]: don-t we run the test with a `timeout` flag?

Comment (aojea) [reply test/integration/framework/etcd.go:132]: duh, this is the RunCustomEtcd, the tests on the jobs run the external etcd

**Pull request closed without merging**